File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.cpp
class Solution {
public:
    string decodeCiphertext(string encodedText, int rows) {
        if(encodedText == ""){
            return "";
        }

        int cols = encodedText.length() / rows;

        vector<vector<char>> mat(rows, vector<char>(cols));

        int ind = 0;

        for(int i = 0; i < rows; i++){
            for(int j = 0; j < cols; j++){
                mat[i][j] = encodedText[ind];
                ind++;
            }
        }

        encodedText = "";

        for(int i = 0; i < cols; i++){
            int r = 0, c = i;
            while(r < rows && c < cols){
                encodedText += mat[r][c];
                r++;
                c++;
            }
        }

        int len = encodedText.length();
        ind = 0;
        for(int i = len - 1; i >= 0; i--){
            if(encodedText[i] == ' '){
                ind++;
            }else{
                break;
            }
        }

        encodedText = encodedText.substr(0,len - ind);
        
        
        return encodedText;
    }
};

File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.py
class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        if encodedText == "":
            return ""
        
        cols = int(len(encodedText) / rows)

        mat = [[None]*cols for _ in range(rows)]
        idx = 0
        for i in range (rows):
            for j in range(cols):
                mat[i][j] = encodedText[idx]
                idx = idx + 1
        
        res = []

        idx = 0

        for i in range(cols):
            r,c = 0,i
            while (r < rows and c < cols):
                res.append(mat[r][c])
                r = r + 1
                c = c + 1
        
        return "".join(res).rstrip()